src/runtime/plugins/supabase.client.ts

```diff
@@ -14,32 +14,34 @@ export default defineNuxtPlugin({
 
     // Handle auth event client side
     supabaseClient.auth.onAuthStateChange(async (event, session) => {
-      if (session) {
-        if (JSON.stringify(user.value) !== JSON.stringify(session.user)) {
-          user.value = session.user;
-        }
-      } else {
-        user.value = null
-      }
-
       if (event === 'SIGNED_IN' || event === 'TOKEN_REFRESHED') {
+        if (session?.user) {
+          if (JSON.stringify(user.value) !== JSON.stringify(session.user)) {
+            user.value = session.user
+          }
+        }
         useCookie(`${cookieName}-access-token`, cookieOptions).value = session?.access_token
         useCookie(`${cookieName}-refresh-token`, cookieOptions).value = session?.refresh_token
-        if (session.provider_token) useCookie(`${cookieName}-provider-token`, cookieOptions).value = session.provider_token
-        if (session.provider_refresh_token) useCookie(`${cookieName}-provider-refresh-token`, cookieOptions).value = session.provider_refresh_token
       }
       if (event === 'SIGNED_OUT') {
+        user.value = null
         useCookie(`${cookieName}-access-token`, cookieOptions).value = null
         useCookie(`${cookieName}-refresh-token`, cookieOptions).value = null
-        useCookie(`${cookieName}-provider-token`, cookieOptions).value = null
-        useCookie(`${cookieName}-provider-refresh-token`, cookieOptions).value = null
+      }
+      if (event === 'USER_UPDATED') {
+        if (session?.user) {
+          user.value = session.user
+        }
       }
     })
 
+    // attempt to retrieve an existing session from local storage
+    await supabaseClient.auth.getSession()
```

Comment:
This call fix the client rendered page issue since it ensures the session/user is set on client side refresh. It means we need to update the `confirm.vue` page in the playground and the documentation (also mention it in the release):

```
const user = useSupabaseUser()
if (user.value) {
  await navigateTo('/user')
}
```